### PR TITLE
API Streamer docs

### PR DESCRIPTION
**Pull request by Srushtika**
This is a WIP for API Streamer docs

### CODE REVIEW

**Comment by Srushtika** [comment]
Hi @remotekieran! This is ready for edits now. 

Some specific things to keep in mind:

1. Accuracy of the copy
2. Completeness of the copy, to push this out publicly with whatever is currently present (i.e. even without the tutorials and such)
3. The logical flow of the copy

File: content/api-streamer/overview.textile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+If you already have a DSN app, you can skip to learning about the "various elements of the API Streamer":#api-streamer-elements and how to set those up.
+
+<%= partial partial_version('/api-streamer/_dsn-app-setup') %>
+
+h2(#api-streamer-elements). Understanding the elements of the API Streamer
+
+With the API Streamer, you can either publish your realtime data streams via your realtime APIs or consume a stream of data by subscribing to it, provided you have the right access.
+
+h3(#producer). Being a producer
```
Comment (mattheworiordan) [inline content/api-streamer/overview.textile]: Perhaps make Producer capital.  My thinking is that it's a role, not just a word.

File: content/api-streamer/overview.textile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+A producer of data uses API Streamer to share live data streams with subscribers. Examples of realtime data streams include live sports scores, positioning updates for transport and logistics, or live data streamed from IoT sensors. These data streams can easily be published to the API Streamer via one of your DSN apps, which can then be shared with "data consumers":#consumer. "Learn more about publishing data streams using the API Streamer":/api-streamer/producer.
```
Comment (mattheworiordan) [inline content/api-streamer/overview.textile]: I think we can eleborate a little more about what API Streamer provides i.e. A producer of data uses API Streamer to share live data streams with subscribers.  The API Streamer allows producers to publish their data once to the Ably DSN, and control who has access to their data without worrying about the complexity of which push or pull protocols consumers are using, or indeed how much data is being consumed.

File: content/api-streamer/overview.textile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+h3(#consumer). Being a consumer
+
+A consumer is usually invited to API Streamer and granted access to a data stream by a producer. The consumer can subscribe to, manage, and consume the data streams. "Learn more about consuming the data streams using the API Streamer":api-streamer/consumer.
```
Comment (mattheworiordan) [inline content/api-streamer/overview.textile]: How is subscribe & consume different? I know we potentially see those as separate operations, but a reader won't.  I also think that we should mention that consumers can choose which pull protocols they wish to use such as MQTT or SSE, or which push protocols they would like to use such as AMQP, Webhooks, Serverless functions or or Kafka.

File: content/api-streamer/overview.textile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+h3(#tutorials). Next up
```
Comment (mattheworiordan) [inline content/api-streamer/overview.textile]: Why is this hash tag tutorials?

File: content/api-streamer/overview.textile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+If you'd like to try out the API Streamer, you can read more about "producers":/api-streamer/producer and "consumers":/api-streamer/consumer or check out our "step-by-step tutorials":todo-link or "video walk-throughs":todo-link to get started quickly.
```
Comment (mattheworiordan) [inline content/api-streamer/overview.textile]: I don't think we should ever release code with `todo-link`s.  This is error prone.  Why not state there are tutorials coming soon instead?

File: content/api-streamer/producer.textile [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+API Streamer is designed for ease of publishing, eliminating the need for custom-built integrations. You can simply publish realtime data to the platform with end-to-end delivery guarantees. Whether your data reaches one or a million subscribers, engineering effort is minimal. 
+
+<!-- INSERT IMAGE -->
+
+h2(#getting-started-as-a-producer). Getting started
+
+Our API Streamer Producer documentation is coming soon... 
+
+If you'd like to get any questions answered before that, "we'll be happy to help":mailto:support@ably.io.
```
Comment (mattheworiordan) [inline content/api-streamer/producer.textile]: You should be linjking to www.ably.io/support.  Please don't use email addresses as the support page gives them many options.

File: content/api-streamer/producer.textile [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+index: 67
+---
+
+A Producer of data uses the "API Streamer":https://ably.io/api-streamer to share live data streams with subscribers. Examples of realtime data streams include live sports scores, positioning updates for transport and logistics, or live data streamed from IoT sensors. These data streams can easily be published to the API Streamer via an Ably DSN app, which can be shared with "data consumers":/api-streamer/consumer. 
+
+You can use the API Streamer to "deploy":#deploy, "manage":#manage and "distribute":#distribute your data streams hosted using your DSN apps.
+
+h3(#deploy). Deploy
+
+Built on our globally-distributed, serverless "Data Stream Network":https://www.ably.io/platform, the API Streamer lets you reliably deploy your streaming APIs. Issues typically encountered when deploying realtime APIs such as scalability, bursts of usage, uptime guarantees, message ordering and data integrity, are all taken care of.
```
Comment (mattheworiordan) [inline content/api-streamer/producer.textile]: The things you are talking about are distribution issues, not deploy issues.  I assume @remotekieran for Deploy, we are thinking about deploying data i.e. data flowing from producer to the DSN right?  Then manage is control of those data flows, and distribute is delivery of the data.

File: content/api-streamer/producer.textile [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+h3(#deploy). Deploy
+
+Built on our globally-distributed, serverless "Data Stream Network":https://www.ably.io/platform, the API Streamer lets you reliably deploy your streaming APIs. Issues typically encountered when deploying realtime APIs such as scalability, bursts of usage, uptime guarantees, message ordering and data integrity, are all taken care of. 
+
+<!-- INSERT IMAGE -->
+
+h3(#manage). Manage
+
+Use API Streamer’s realtime native API Management toolkit to maximize the value of your realtime APIs. Streamer’s developer portal gives you security, analytics, stream transformation, as well as monetization and billing tools. Streamer helps turn data streams into revenue streams.
```
Comment (mattheworiordan) [inline content/api-streamer/producer.textile]: I don't know what you mean by "realtime native API management", can you explain?

File: content/api-streamer/producer.textile [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+h3(#manage). Manage
+
+Use API Streamer’s realtime native API Management toolkit to maximize the value of your realtime APIs. Streamer’s developer portal gives you security, analytics, stream transformation, as well as monetization and billing tools. Streamer helps turn data streams into revenue streams. 
+
+<!-- INSERT IMAGE -->
+
+h3(#distribute). Distribute
+
+API Streamer is designed for ease of publishing, eliminating the need for custom-built integrations. You can simply publish realtime data to the platform with end-to-end delivery guarantees. Whether your data reaches one or a million subscribers, engineering effort is minimal.
```
Comment (mattheworiordan) [inline content/api-streamer/producer.textile]: No mention of protocols, and listing them out, which is arguably the most compellling part of this offering.

File: content/api-streamer/consumer.textile [hunk 10]
```diff
@@ -0,0 +1,34 @@
+---
+title: Consumer
+section: api-streamer
+index: 68
+---
+
+API Streamer has two roles: a producer and consumer. A consumer is usually invited to API Streamer by a producer so the consumer can subscribe to, manage, and consume the data streams. A data stream could be anything from live transport updates to a stream of sensor data from an embedded IoT device.
```
Comment (mattheworiordan) [inline content/api-streamer/consumer.textile]: > A consumer is usually invited to API Streamer by a producer so the consumer can subscribe to, manage, and consume the data streams

I don't like this "usually invited" part.  What happens the rest of the time.

"A consumer subscribes to streams offered by producers using any of the many open push and pull protocols supported by Ably."  I am not really sure how subscribe/consume is different, or why management is important for consumers

File: content/api-streamer/consumer.textile [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+* significantly lowers integration time and engineering cost. You can use the protocols or adapters available to simplify data ingestion.
+
+* allows interoperability for open streaming protocols from SSE, to MQTT, to STOMP.
+
+* allows APIs to be event-driven, capable of pushing data directly into queueing or data processing engines, legacy systems, or even trigger serverless functions in the cloud. For example, Apache Kafka, Amazon Kinesis, or Microsoft Azure.
+
+* offers an end-to-end solution for delivering data to systems or even end-user devices. With current APIs, consumers must process data from producers, store it, and publish it for their end users. API Streamer can enable delivery of data from source to subscriber.
+
+* does away with polling data. Pub/Sub architecture provides high-performance realtime APIs where new updates are pushed directly to you without you having to pull them.
```
Comment (mattheworiordan) [inline content/api-streamer/consumer.textile]: This section feels like a marketing section, yet we have landing pages.  I thought the documentation is aimed at developers, focussing on tangible (non opinionated) benefits, and how they get started with the API streamer.  I fear this documentation is some times too high level, and often marketing focussed.  Our documentation needs to give **developers** an overview of concepts / the service, provide the detail they need to understand what we offer (they are techies) and give them the info they need to get started now.

File: content/api-streamer/producer.textile (hunk 8 above)
Comment (Srushtika) [reply content/api-streamer/producer.textile]: ^ @remotekieran

**Comment by mattheworiordan** [comment]
@Srushtika is this going to be deployed as a new branch?  Why was this closed?

Also, please be careful with your git commit history...  What are conflict commits for, and a bunch of reverts, and the WIP commit from me shoudl have been edited and landed (it was a WIP commit i.e. no commits should be put on top of that in reality)

screenshot_2019-08-12_03-57-30_pm

**Comment by Srushtika** [comment]
@mattheworiordan there was some weird stuff that was being shown as something I committed, I along with a bit of Tom's help tried to revert the changes to go back to your version but only made it worse. After everything, this branch still had a lot of conflicts with the master as well as this a very old one, it felt easier to create a new branch and PR: https://github.com/ably/docs/pull/716 

Three of the commits you did in this PR were not related to API Streamer specifically but to docs in general, so committed them directly to master separately: https://github.com/ably/docs/commits/master

Apologies, I should have mentioned that at the time of closing this PR.

**Comment by mattheworiordan** [comment]
> there was some weird stuff that was being shown as something I committed, 

Ok, well it was done, but you should definitely get a handle on how to handle these types of conflicts as it's a core skill in coding (source control management, rebasing, handling conflicts, fixup commits, editing old commits etc).

> I along with a bit of Tom's help tried to revert the changes to go back to your version but only made it worse.

Why not just `git reset [SHA] --hard` instead of reverts?

> After everything, this branch still had a lot of conflicts with the master as well as this a very old one, it felt easier to create a new branch and PR

Ok, fair enough, but this is a "smell" to be honest and something we should learn to avoid.  I would help you, but hard after the fact now :(

> Three of the commits you did in this PR were not related to API Streamer specifically but to docs in general, so committed them directly to master separately: 

Yes, I know, my bad. Sorry.

**Pull request closed without merging**